public/app/plugins/datasource/azuremonitor/resourcePicker/resourcePickerData.ts

```diff
@@ -81,28 +81,84 @@ export default class ResourcePickerData extends DataSourceWithBackend<
     return resources;
   }
 
+  async fetchFiltered(
+    type: ResourcePickerQueryType,
+    filters: { subscriptions: string[]; types: string[]; locations: string[] },
+    currentSelection?: AzureMonitorResource[]
+  ): Promise<ResourceRowGroup> {
+    try {
```

Comment:
When this function is called there is a try-catch around it. Should we refactor this code to not have the try-catch?

Reply:
We could move the error checking to the function at the higher level, btu we'd then have to ensure we have the same error checking wherever this is called if it's used in the future which might make it a bit unclear.